packages/zosfiles/__tests__/__system__/cli/create/zfs/cli.files.create.zfs.system.test.ts

```diff
@@ -0,0 +1,107 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+import { Session } from "@zowe/imperative";
+import { getUniqueDatasetName, runCliScript } from "../../../../../../../__tests__/__src__/TestUtils";
+import { TestEnvironment } from "../../../../../../../__tests__/__src__/environment/TestEnvironment";
+import { ITestEnvironment } from "../../../../../../../__tests__/__src__/environment/doc/response/ITestEnvironment";
+import { ITestPropertiesSchema } from "../../../../../../../__tests__/__src__/properties/ITestPropertiesSchema";
+import { Delete } from "../../../../../src/api/methods/delete";
+
+const ZOWE_OPT_BASE_PATH = "ZOWE_OPT_BASE_PATH";
+
+let REAL_SESSION: Session;
+// Test Environment populated in the beforeAll();
+let TEST_ENVIRONMENT: ITestEnvironment;
+let TEST_ENVIRONMENT_NO_PROF: ITestEnvironment;
+let defaultSystem: ITestPropertiesSchema;
+let fsname: string;
+let volume: string;
+
+describe("Create z/OS File System", () => {
+
+    // Create the unique test environment
+    beforeAll(async () => {
+        TEST_ENVIRONMENT = await TestEnvironment.setUp({
+            tempProfileTypes: ["zosmf"],
+            testName: "zos_create_zos_file_system"
+        });
+
+        defaultSystem = TEST_ENVIRONMENT.systemTestProperties;
+
+        REAL_SESSION = TestEnvironment.createZosmfSession(TEST_ENVIRONMENT);
+
+        fsname = getUniqueDatasetName(defaultSystem.zosmf.user);
+        volume = defaultSystem.datasets.vol;
+    });
+
+    afterAll(async () => {
+        await TestEnvironment.cleanUp(TEST_ENVIRONMENT);
+    });
+
+    describe("without profiles", () => {
+        let defaultSys: ITestPropertiesSchema;
+
+        // Create the unique test environment
+        beforeAll(async () => {
+            TEST_ENVIRONMENT_NO_PROF = await TestEnvironment.setUp({
+                testName: "zos_files_create_zos_file_system_without_profile"
+            });
+
+            defaultSys = TEST_ENVIRONMENT_NO_PROF.systemTestProperties;
+        });
+
+        afterEach(async () => {
```

Comment:
I think its OK as is, but as a side note... if a test process dies or is killed during a create test, the subsequent run may fail. You could ensure that the environment is "ready/prepped" in beforeAll by deleting any residual "stuff" from a previous run.

Reply:
Not sure how to cleanup residual stuff from previous run in this case, because the `fileSystemName` gets pseudorandomly generated. Therefore the name of the ZFS to delete is only known in the current test environment.

Reply:
Probably not possible then in this case 😎